device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -601,6 +625,12 @@ public void unregisterDeviceClients(Iterable<DeviceClient> deviceClients, long t
                 log.info("Unregistering device {} from multiplexing client", deviceClientToUnregister.getConfig().getDeviceId());
                 this.multiplexedDeviceClients.remove(deviceClientToUnregister.getConfig().getDeviceId());
                 deviceClientToUnregister.setDeviceIO(null);
+
+                // Clear the device client's subscriptions after it has been unregistered from the multiplexing client
+                // We may make this optional at some point in the future if users want to be able to preserve subscriptions.
+                deviceClientToUnregister.markTwinAsUnsubscribed();
+                deviceClientToUnregister.markMethodsAsUnsubscribed();
+                deviceClientToUnregister.setMessageCallback(null, null);
```

Comment:
MessageCallback allows for us and users to set the callback to null, but methods and twin do not, hence the need for these new markTwin/MethodAsUnsubscribed methods